lib/pure/streams.nim

```diff
@@ -224,10 +224,12 @@ proc ssReadData(s: Stream, buffer: pointer, bufLen: int): int =
 
 proc ssWriteData(s: Stream, buffer: pointer, bufLen: int) = 
   var s = StringStream(s)
-  if bufLen > 0: 
+  if bufLen <= 0:
+    return
+  if s.pos + bufLen > s.data.len():
     setLen(s.data, s.data.len + bufLen)
```

Comment:
This makes the underlying string too big actually, doesn't it? Shouldn't that be `setLen(s.data, s.pos + bufLen)`?